File: library/src/main/java/me/proxer/library/entity/messenger/ConferenceInfo.java
package me.proxer.library.entity.messenger;

import com.squareup.moshi.Json;
import lombok.EqualsAndHashCode;
import lombok.Value;

import javax.annotation.Nullable;
import java.util.Date;
import java.util.List;

/**
 * Entity that holds various info of a {@link Conference}.
 *
 * @author Desnoo
 */
@Value
@EqualsAndHashCode(onParam = @__({@Nullable}))
public class ConferenceInfo {

    /**
     * Returns the topic.
     */
    @Json(name = "topic")
    private String topic;

    /**
     * Returns the amount of participants.
     */
    @Json(name = "count")
    private int participantAmount;

    /**
     * Returns the time, the first message was sent.
     */
    @Json(name = "timestamp_start")
    private Date firstMessageTime;

    /**
     * Returns the time, the last message was sent.
     */
    @Json(name = "timestamp_end")
    private Date lastMessageTime;

    /**
     * Returns the id of the conference leader.
     */
    @Json(name = "leader")
    private String leaderId;

    /**
     * Returns the participants of the conference.
     */
    @Json(name = "users")
    private List<ConferenceParticipant> participants;
}

File: library/src/main/kotlin/me/proxer/library/entity/messenger/ConferenceInfo.kt
package me.proxer.library.entity.messenger

import com.squareup.moshi.Json
import java.util.Date

/**
 * Entity that holds various info of a [Conference].
 *
 * @property topic The topic.
 * @property participantAmount The amount of participants.
 * @property firstMessageTime The time, the first message was sent.
 * @property lastMessageTime The time, the last message was sent.
 * @property leaderId The id of the conference leader.
 * @property participants The participants of the conference.
 *
 * @author Desnoo
 */
data class ConferenceInfo(
    @Json(name = "topic") val topic: String,
    @Json(name = "count") val participantAmount: Int,
    @Json(name = "timestamp_start") val firstMessageTime: Date,
    @Json(name = "timestamp_end") val lastMessageTime: Date,
    @Json(name = "leader") val leaderId: String,
    @Json(name = "users") val participants: List<ConferenceParticipant>
)
